### ISSUE TITLE
is `include_intercept = TRUE` working as intended in `eta_squared()`?

### ISSUE THREAD

**Issue by cedricbatailler** [easystats/effectsize#438]
**Question and context**

I am a but surprised by the behavior of `include_intercept = TRUE` in `eta_squared` which does not appear to change anything. 

``` r
fit <- lm(mpg ~ cyl, mtcars)

effectsize::eta_squared(
  fit, 
  include_intercept = TRUE
)
#> For one-way between subjects designs, partial eta squared is equivalent to eta squared.
#> Returning eta squared.
#> # Effect Size for ANOVA
#> 
#> Parameter | Eta2 |       95% CI
#> -------------------------------
#> cyl       | 0.73 | [0.57, 1.00]
#> 
#> - One-sided CIs: upper bound fixed at [1.00].
```

Here, I would have expected an additional line with the partial eta squared corresponding to the intercept (i.e., difference of explained variance between `mpg ~ cyl` and `mpg ~ 0 + cyl`). However, it does not seem to be the case. Am I missing something?

Using the dev version of {effectsize} if that's important.

**Comment by mattansb** [comment]
Interesting! I think this will only happen for a single term fixed effects model - oops!

**Comment by mattansb** [comment]
Okay, so this actually isn't a bug (at least not in `effectsize`).

The SS for the effect sizes are taken from the `anova()` function, which does not return SS for the intercept:

``` r
fit <- lm(mpg ~ cyl + am, mtcars)

anova(fit)
#> Analysis of Variance Table
#> 
#> Response: mpg
#>           Df Sum Sq Mean Sq F value    Pr(>F)    
#> cyl        1 817.71  817.71 87.3876 2.977e-10 ***
#> am         1  36.97   36.97  3.9511   0.05635 .  
#> Residuals 29 271.36    9.36                      
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
```

To obtain the intercept, you need to pass a model or ANOVA table that has that information.
For example:

``` r
(A1 <- car::Anova(fit, type = 3)) # type = 2 does not return this!
#> Anova Table (Type III tests)
#> 
#> Response: mpg
#>              Sum Sq Df  F value    Pr(>F)    
#> (Intercept) 1645.68  1 175.8706 7.694e-14 ***
#> cyl          449.53  1  48.0410 1.285e-07 ***
#> am            36.97  1   3.9511   0.05635 .  
#> Residuals    271.36 29                       
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

effectsize::eta_squared(
  A1, 
  include_intercept = TRUE
)
#> # Effect Size for ANOVA (Type III)
#> 
#> Parameter   | Eta2 (partial) |       95% CI
#> -------------------------------------------
#> (Intercept) |           0.86 | [0.77, 1.00]
#> cyl         |           0.62 | [0.43, 1.00]
#> am          |           0.12 | [0.00, 1.00]
#> 
#> - One-sided CIs: upper bound fixed at [1.00].
```

I’ve added a little message for when `include_intercept = TRUE` but the SS cannot be found.

``` r
effectsize::eta_squared(
  fit, 
  include_intercept = TRUE
)
#> Could not find Sum-of-Squares for the (Intercept) in the ANOVA table.
#> # Effect Size for ANOVA (Type I)
#> 
#> Parameter | Eta2 (partial) |       95% CI
#> -----------------------------------------
#> cyl       |           0.75 | [0.61, 1.00]
#> am        |           0.12 | [0.00, 1.00]
#> 
#> - One-sided CIs: upper bound fixed at [1.00].
```

<sup>Created on 2022-05-03 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
